### PR TITLE
Change parseEnum to something faster for method parsing in asynchttpserver

### PR DESCRIPTION
**Pull request by hlaaftana**

### CODE REVIEW

File: lib/pure/asynchttpserver.nim [hunk 1]
```diff
@@ -132,6 +132,48 @@ proc parseProtocol(protocol: string): tuple[orig: string, major, minor: int] =
 proc sendStatus(client: AsyncSocket, status: string): Future[void] =
   client.send("HTTP/1.1 " & status & "\c\L\c\L")
 
+proc parseUppercaseMethod(name: string): HttpMethod =
```
Comment (genotrance) [inline lib/pure/asynchttpserver.nim]: Cool code but how much faster is this compared to a readable `name == "POST"`?

**Comment by dom96** [comment]
I'd prefer it if ``asynchttpserver`` remains readable, even at the expense of speed. I'm writing a performant implementation here: https://github.com/dom96/httpbeast

**Comment by cooldome** [comment]
It is possible to improve performance of parseEnum itself

**Comment by hlaaftana** [comment]
Even if parseEnum was improved the string concatenation with "http" would still have overhead. In that case a case statement would be the way to go, so I'll change it to that and this PR could be squashed if the result was desirable.

File: lib/pure/asynchttpserver.nim [hunk 2]
```diff
@@ -132,6 +132,19 @@ proc parseProtocol(protocol: string): tuple[orig: string, major, minor: int] =
...
+  result = case name
+  of "GET": HttpGet
```
Comment (Araq) [inline lib/pure/asynchttpserver.nim]: Please align the ``of`` under the ``case`` keyword.

Comment (dom96) [reply lib/pure/asynchttpserver.nim]: To be clear:

```nim
result =
  case name
  of "GET": ...
```

**Pull request merged**